File: Java/DesignPatterns/src/test/java/designPatterns/structural/bridge/MainBridgeTest.java
package designPatterns.structural.bridge;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainBridgeTest {

  private Abstraction abs1;
  private Abstraction abs2;

  @BeforeEach
  void setUp() {
    abs1 = new Refined(new ConcreteImplementor.A());
    abs2 = new Refined(new ConcreteImplementor.B());
  }

  @Test
  void equalsTest() {
    assertThat(abs1.doSomething()).isEqualTo("Refined: Applying A");
    assertThat(abs2.doSomething()).isEqualTo("Refined: Applying B");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/structural/bridge/MainBridgeTest.kt
package structural.bridge

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class MainBridgeTest {
  private lateinit var abs1: Abstraction
  private lateinit var abs2: Abstraction

  @BeforeEach
  internal fun setUp() {
    abs1 = Refined(ConcreteImplementor.A())
    abs2 = Refined(ConcreteImplementor.B())
  }

  @Test
  internal fun equalsTest() {
    assertThat(abs1.doSomething()).isEqualTo("Refined: Applying A")
    assertThat(abs2.doSomething()).isEqualTo("Refined: Applying B")
  }
}